### PR TITLE
Mongodb docker

### PR DESCRIPTION
**Pull request by jstucke**

### CODE REVIEW

File: src/install/db.py [hunk 1]
```diff
@@ -26,31 +14,16 @@ def _get_db_directory():
...
+def main():
     logging.info('Setting up mongo database')
-
-    if distribution in ['xenial', 'debian']:
-        _add_mongo_mirror(distribution)
-        apt_update_sources()
-        apt_install_packages('mongodb-org')
-    else:
-        apt_install_packages('mongodb')
+    _, return_code = execute_shell_command_get_return_code('docker pull mongo:3-xenial')
```
Comment (weidenba) [inline src/install/db.py:19]: Why xenial with EOL next year?

File: src/storage/mongodb_docker.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+def get_docker_network(docker_client: DockerClient) -> Network:
+    try:
+        network = docker_client.networks.get(FACT_MONGODB_NETWORK)
+    except NotFound:
+        network = create_docker_network(docker_client)
+    return network
+
+
+def create_docker_network(client: DockerClient) -> Network:
+    ipam_pool = docker.types.IPAMPool(subnet='192.168.27.0/24', gateway='192.168.27.1')
```
Comment (weidenba) [inline src/storage/mongodb_docker.py:87]: can this interfere with other local networks? Should we make this IP-Adresses configureable somehow?

File: src/install/db.py (hunk 1 above)
Comment (jstucke) [reply src/install/db.py:19]: sadly newer versions like bionic are only available with MongoDB 4
I tried to build a custom container but was not able to connect to it (for still unknown reasons)

File: src/storage/mongodb_docker.py (hunk 2 above)
Comment (jstucke) [reply src/storage/mongodb_docker.py]: made ip, subnet and gateway configurable

File: src/install/db.py (hunk 1 above)
Comment (jstucke) [reply src/install/db.py]: replaced with a docker image based on debian stretch

**Comment by jstucke** [comment]
replaced by https://github.com/fkie-cad/FACT_docker

**Pull request closed without merging**